src/fides/api/ops/models/privacy_notice.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from enum import Enum
+from typing import Any, Dict, Iterable, List, Tuple
+
+from sqlalchemy import Boolean, Column
+from sqlalchemy import Enum as EnumColumn
+from sqlalchemy import Float, ForeignKey, String
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.orm import Session, relationship
+
+from fides.api.ops.common_exceptions import ValidationError
+from fides.lib.db.base_class import Base, FidesBase
+
+
+class PrivacyNoticeRegion(Enum):
+    """
+    Enum is not formalized in the DB because it is subject to frequent change
+    """
+
+    us_ca = "us_ca"
+    us_co = "us_co"
+    us_va = "us_va"
+    us_ut = "us_ut"
+    eu = "eu"
+
+
+class ConsentMechanism(Enum):
+    opt_in = "opt_in"
+    opt_out = "opt_out"
+    necessary = "necessary"
+
+
+class EnforcementLevel(Enum):
+    """
+    Enum is not formalized in the DB because it may be subject to frequent change
+    """
+
+    frontend = "frontend"
+    system_wide = "system_wide"
+
+
+class PrivacyNoticeBase:
+    """
+    Base class to establish the common columns for `PrivacyNotice`s and `PrivacyNoticeHistory`s
+    """
+
+    name = Column(String, nullable=False)
+    description = Column(String, nullable=True)
+    origin = Column(String, nullable=True)  # pointer back to an origin template ID
+    regions = Column(
+        ARRAY(EnumColumn(PrivacyNoticeRegion, native_enum=False)),
+        index=True,
+        nullable=False,
+    )
+    consent_mechanism = Column(EnumColumn(ConsentMechanism), nullable=False)
+    data_uses = Column(
+        ARRAY(String), nullable=False
+    )  # a list of `fides_key`s of `DataUse` records
+    enforcement_level = Column(EnumColumn(EnforcementLevel), nullable=False)
+    version = Column(Float, nullable=False, default=1.0)
+    disabled = Column(Boolean, nullable=False, default=False)
+    has_gpc_flag = Column(Boolean, nullable=False, default=False)
+    displayed_in_privacy_center = Column(Boolean, nullable=False, default=True)
+    displayed_in_banner = Column(Boolean, nullable=False, default=True)
+    displayed_in_privacy_modal = Column(Boolean, nullable=False, default=True)
+
+
+class PrivacyNotice(PrivacyNoticeBase, Base):
```

Comment:
was thinking i should probably add some unit tests specifically around the methods in this class? my general endpoint testing does seem to provide good coverage (at least codecov isn't complaining), but those are more "end to end" and it may be better to break down some of the testing to target these components specifically...